solenoid/records/tests/test_views.py

```diff
@@ -0,0 +1,319 @@
+    """
+    If we re-import an already sent record with a known ID & author, we
+    should raise a warning and leave the existing record alone, not create
+    a new one."""
+    orig_count = Record.objects.count()
+
+    with pytest.raises(Record.DoesNotExist):
+        Record.objects.get(paper_id='12345')
+
+    client.post(IMPORT_URL, {'author_id': '98765'})
+    new_count = Record.objects.count()
+    assert orig_count + 1 == new_count
+
+    record = Record.objects.latest('pk')
+    assert '12345' == record.paper_id
+
+    liaison = Liaison.objects.create(first_name='foo',
+                                     last_name='bar',
+                                     email_address='fake@example.com')
+
+    email = EmailMessage.objects.create(original_text='gjhdka',
+                                        date_sent=date.today(),
+                                        author=record.author,
+                                        _liaison=liaison)
+
+    record.email = email
+    record.save()
+    orig_record = model_to_dict(record)
+
+    response = client.post(IMPORT_URL, {'author_id': '98765'}, follow=True)
+    assert new_count == Record.objects.count()
+    assert orig_record == model_to_dict(Record.objects.get(paper_id='12345'))
+    assert 'info' in [m.level_tag for m in response.context['messages']]
+
+
+@pytest.mark.django_db
+def test_paper_id_respected_case_4(client, mock_elements, test_settings):
+    """
+    If we re-import an already sent record with a known ID and a new
+    author, we should raise a warning and not create a new record."""
+    orig_count = Record.objects.count()
+    with pytest.raises(Record.DoesNotExist):
+        Record.objects.get(paper_id='12345')
+    client.post(IMPORT_URL, {'author_id': '98765'})
+
+    new_count = Record.objects.count()
+    assert orig_count + 1 == new_count
+
+    record = Record.objects.latest('pk')
+    assert '12345' == record.paper_id
+
+    liaison = Liaison.objects.create(first_name='foo',
+                                     last_name='bar',
+                                     email_address='fake@example.com')
+
+    email = EmailMessage.objects.create(original_text='gjhdka',
+                                        date_sent=date.today(),
+                                        author=record.author,
+                                        _liaison=liaison)
+    record.email = email
+    record.save()
+    orig_record = model_to_dict(record)
+
+    # Next, import new data about the same paper as recorded under a different
+    # author. This should *not* create any new records.
+    response = client.post(IMPORT_URL, {'author_id': '54321'}, follow=True)
+    assert new_count == Record.objects.count()
+    assert orig_record == model_to_dict(Record.objects.get(paper_id='12345'))
+    assert 'info' in [m.level_tag for m in response.context['messages']]
+
+
+# TODO: check with stakeholders to find out if any of the below concerns are
```

Comment:
You might just want to mark these tests as skipped, rather than commenting out. I'd be likely to forget I commented them out, but if they constantly show up as skipped when I run tests it's harder to ignore.

Reply:
Good suggestion, thanks!